app/javascript/components/assessment/steps/SymptomsAssessment.js

```diff
@@ -77,19 +98,41 @@ class SymptomsAssessment extends React.Component {
   };
 
   handleSubmit = async () => {
+    const reportState = this.formatedReportState();
     if (this.fieldIsEmptyOrNew(this.props.assessment)) {
-      this.props.submit(this.state.reportState);
+      this.props.submit(reportState);
     } else {
       if (this.hasChanges()) {
         if (await confirmDialog("Are you sure you'd like to modify this report?")) {
-          this.props.submit(this.state.reportState);
+          this.props.submit(reportState);
         } else {
           this.setState({ loading: false });
         }
       } else {
-        this.props.submit(this.state.reportState);
+        this.props.submit(reportState);
+      }
+    }
+  };
+
+  // Converts all FloatSymptoms and IntergerSymptoms to numerical values and nulls out any non-numerical values provided (such as '-', '.', and '-.')
+  formatedReportState = () => {
+    let reportState = this.state.reportState;
+    for (const key in this.state.reportState['symptoms']) {
+      if (parseInt(key) && reportState['symptoms'][parseInt(key)].type == 'FloatSymptom') {
+        if (!isNaN(parseFloat(reportState['symptoms'][parseInt(key)].value))) {
+          reportState['symptoms'][parseInt(key)].value = parseFloat(reportState['symptoms'][parseInt(key)].value);
```

Comment:
I'm a little confused about the need for this function. Shouldn't this parsing be handled in the change handlers for each of these types above? So that it's already in the correct format when it gets set in state in the first place.

Reply:
The reason I needed to create this method was for the case where the user inputs a value such as `-`, `.`, or `-.` without any associated digit and submits the assessment. The text box needs to accept these values so the user can type in a negative or decimal number if needed which is why the change method accepts them as valid inputs. So if I don't have this formatting function, then an error will occur if a user happens to submit the assessment with one of the values I listed above. This formatting function ensures that if such an input is provided, it is nulled out so such an error does not occur.

Reply:
Wouldn't the `isNaN()` check in the handlers take care of that though?

Reply:
In the change handler, there is a `isNaN()` check, but it is included in an 'Or' condition that also allows the user to type the input if it is a `.`, `-`, or `-.` without the input needing to pass the `isNaN()` check. Refer to the last portion of the condition statement on [this line](https://github.com/SaraAlert/SaraAlert/blob/1145-input-validation/app/javascript/components/assessment/steps/SymptomsAssessment.js#L58).

Or perhaps I am misunderstanding your question?

Reply:
Resolved internally - I was forgetting these handlers are called per character.